colcon_core/verb/build.py

```diff
@@ -116,6 +123,7 @@ def add_arguments(self, *, parser):  # noqa: D102
         self.task_argument_destinations = decorated_parser.get_destinations()
 
     def main(self, *, context):  # noqa: D102
+        check_and_mark_root_dir(context.args.start_path)
```

Comment:
I think the invocation here in the verb is too late since e.g. the `log` directory has already been created which isn't desired.